File: kafka-streams/src/test/groovy/io/micronaut/configuration/kafka/streams/wordcount/WordCountClient.java
package io.micronaut.configuration.kafka.streams.wordcount;

import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.configuration.kafka.annotation.Topic;

@KafkaClient
public interface WordCountClient {

    @Topic(WordCountStream.INPUT)
    void publishSentence(String sentence);
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/streams/WordCountClient.kt
package io.micronaut.kafka.docs.streams

// tag::imports[]
import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.configuration.kafka.annotation.Topic
import io.micronaut.context.annotation.Requires
// end::imports[]

@Requires(property = "spec.name", value = "WordCountStreamTest")
// tag::clazz[]
@KafkaClient
interface WordCountClient {
    @Topic(WordCountStream.INPUT)
    fun publishSentence(sentence: String?)
}
// tag::clazz[]
